ddtrace/tracer/span.go

```diff
@@ -319,13 +332,94 @@ func (s *span) finish(finishTime int64) {
 	}
 	s.finished = true
 
+	if t, ok := internal.GetGlobalTracer().(*tracer); ok {
+		// we have an active tracer
+		feats := t.features.Load()
+		if feats.Stats && shouldComputeStats(s) {
+			// the agent supports computed stats
+			select {
+			case t.stats.In <- newAggregableSpan(s, t.config.version):
+				// ok
+			default:
+				log.Error("Stats channel full, disregarding span.")
+			}
+		}
+		if feats.DropP0s {
+			// the agent supports dropping p0's in the client
+			if shouldDrop(s) {
+				// ...and this span can be dropped
+				atomic.AddUint64(&t.droppedP0Spans, 1)
+				if s == s.context.trace.root {
+					atomic.AddUint64(&t.droppedP0Traces, 1)
+				}
+				return
+			}
+		}
+	}
 	if s.context.drop {
 		// not sampled by local sampler
 		return
 	}
 	s.context.finish()
 }
 
+// newAggregableSpan creates a new summary for the span s, within an application
+// version version.
+func newAggregableSpan(s *span, version string) *aggregableSpan {
+	var statusCode uint32
+	if sc, ok := s.Meta["http.status_code"]; ok && sc != "" {
+		if c, err := strconv.Atoi(sc); err == nil {
+			statusCode = uint32(c)
+		}
+	}
+	key := aggregation{
+		Name:       s.Name,
+		Resource:   s.Resource,
+		Service:    s.Service,
+		Type:       s.Type,
+		Hostname:   s.Meta[keyHostname],
+		Synthetics: strings.HasPrefix(s.Meta[keyOrigin], "synthetics"),
+		Env:        s.Meta[ext.Environment],
+		StatusCode: statusCode,
+		Version:    version,
+	}
+	return &aggregableSpan{
+		key:      key,
+		Start:    s.Start,
+		Duration: s.Duration,
+		TopLevel: s.Metrics[keyTopLevel] == 1,
+		Error:    s.Error,
+	}
+}
+
+// shouldDrop reports whether it's fine to drop the span s.
+func shouldDrop(s *span) bool {
+	if p, ok := s.context.samplingPriority(); ok && p > 0 {
```

Comment:
do we need a case for app analytics here?

Reply:
yes analytics is covered here: https://github.com/DataDog/dd-trace-go/blob/39c94f03392e009c7da43966412a9f220d32e692/ddtrace/tracer/span.go#L405-L407